File: DSA/Matrix.java
public class Matrix {
    public static void main(String[] args) {
        int[][] matrix = {
            {3, 7, 8, 9},
            {10, 11, 12, 13},
            {14, 15, 16, 17},
            {18, 19, 20, 21}
        };

        printMatrix(matrix);
    }

    public static void printMatrix(int[][] matrix) {
        if (matrix == null || matrix.length == 0 || matrix[0].length == 0) {
            return;
        }

        int rows = matrix.length;
        int cols = matrix[0].length;
        int top = 0, bottom = rows - 1, left = 0, right = cols - 1;

        while (top <= bottom && left <= right) {
           
            for (int i = left; i <= right; i++) {
                System.out.print(matrix[top][i] + " ");
            }
            top++;

            
            for (int i = top; i <= bottom; i++) {
                System.out.print(matrix[i][right] + " ");
            }
            right--;

            
            if (top <= bottom) {
                for (int i = right; i >= left; i--) {
                    System.out.print(matrix[bottom][i] + " ");
                }
                bottom--;
            }

           
            if (left <= right) {
                for (int i = bottom; i >= top; i--) {
                    System.out.print(matrix[i][left] + " ");
                }
                left++;
            }
        }
    }
}

File: DSA/matrix.c

#include <stdio.h>

int main() {
    int matrix[5][5] = {
        {3, 5, 7, 9, 11},
        {13, 15, 17, 19, 21},
        {23, 25, 27, 29, 31},
        {33, 35, 37, 39, 41},
        {43, 45, 47, 49, 51}
    };
    
    int top = 0, bottom = 4, left = 0, right = 4;

    while (top <= bottom && left <= right) {
        for (int i = left; i <= right; i++) printf("%d ", matrix[top][i]);
        top++;
        for (int i = top; i <= bottom; i++) printf("%d ", matrix[i][right]);
        right--;
        if (top <= bottom) {
            for (int i = right; i >= left; i--) printf("%d ", matrix[bottom][i]);
            bottom--;
        }
        if (left <= right) {
            for (int i = bottom; i >= top; i--) printf("%d ", matrix[i][left]);
            left++;
        }
    }

    printf("\n");
    return 0;
}
